.github/workflows/dashboards-observability-test-and-build-workflow.yml

```diff
@@ -9,11 +9,81 @@ env:
   OPENSEARCH_PLUGIN_VERSION: 3.0.0.0
 
 jobs:
+  Get-CI-Image-Tag:
+    uses: opensearch-project/opensearch-build/.github/workflows/get-ci-image-tag.yml@main
+    with:
+      product: opensearch-dashboards
 
-  build:
+  build-linux:
+    needs: Get-CI-Image-Tag
+    strategy: 
+      fail-fast: false
+    runs-on: ubuntu-latest
+    container:
+      # using the same image which is used by opensearch-build team to build the OpenSearch Distribution
+      # this image tag is subject to change as more dependencies and updates will arrive over time
+      image: ${{ needs.Get-CI-Image-Tag.outputs.ci-image-version-linux }}
+      # need to switch to root so that github actions can install runner binary on container without permission issues.
+      options: --user root
+
+    steps:
+      - name: Checkout OpenSearch Dashboards
+        uses: actions/checkout@v2
+        with:
+          repository: opensearch-project/OpenSearch-Dashboards
+          ref: ${{ env.OPENSEARCH_VERSION }}
+          path: OpenSearch-Dashboards
+
+      - name: Checkout Dashboards Observability
+        uses: actions/checkout@v2
+        with:
+          path: OpenSearch-Dashboards/plugins/dashboards-observability
+
+      - name: Plugin Bootstrap
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 20
+          max_attempts: 2
+          command: |
+            chown -R 1000:1000 `pwd`
+            cd ./OpenSearch-Dashboards/
+            su `id -un 1000` -c "source $NVM_DIR/nvm.sh && nvm use && node -v && yarn -v &&
+                                 yarn config set network-timeout 1000000 -g &&
+                                 yarn osd bootstrap"
+
+      - name: Test all dashboards-observability modules
+        run: |
+            chown -R 1000:1000 `pwd`
+            cd ./OpenSearch-Dashboards/
+            su `id -un 1000` -c "source $NVM_DIR/nvm.sh && nvm use && node -v && yarn -v &&
+                                 cd plugins/dashboards-observability &&
+                                 yarn osd bootstrap && yarn test --coverage --maxWorkers=100%"
```

Comment:
We shouldn't need `yarn osd bootstrap` again. If this is done already on line 52.

Reply:
One in OSD folder one in obs folder.
Normally I see doing this in obs folder / plugin folder is enough.
Tho your original code did it in OSD not plugin so it might miss things that is why I add it again, should not harm 😄

Reply:
In my experience, just the OSD bootstrap has sufficed, not sure I've needed two bootstraps before

Reply:
If bootstrap in plugin folder then yeah one is enough.

Reply:
doesn't matter where, it's calling the same `osd` script